Refactor home page structure and improve accessibility

enntang opened the pull request:
Enhance the home page by restructuring components, updating links for better accessibility, and migrating styles to Tailwind. Introduce new animations and improve overall styling while ensuring a smoother user experience.

enntang committed:
new images

enntang committed:
basic structure

enntang committed:
fix(Footer, Keyvision, Navbar): update links and improve accessibility

enntang committed:
feat: refactor home page structure and components

enntang committed:
feat(BaseLayout): restructure home page components and styles

enntang committed:
refactor(BubbleDialog, FunctionCarousel): improve animation handling and scrolling logic

enntang committed:
fix(BubbleDialog): correct logic for text erasure in animation handling

enntang committed:
feat(BaseLayout): enhance styling and structure of home page components, add new animations and color themes

enntang committed:
refactor(BaseLayout): update component styles and structure

enntang committed:
refactor: migrate CSS to Tailwind and update components

- Remove CSS files and migrate to Tailwind classes
- Fix Footer.tsx accessibility issues by providing valid href attributes
- Update social media links with proper URLs and target attributes
- Improve component styling with Tailwind utilities

enntang committed:
refactor(BaseLayout): replace button elements with reusable Button component and update styles

enntang committed:
refactor(VideoSection): clean up imports and standardize className formatting

JohnsonMao commented on app/[language]/(BaseLayout)/SectionHeader/README.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

JohnsonMao commented on app/[language]/(BaseLayout)/SectionHeader/REFACTORING_SUMMARY.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/README.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on app/[language]/(BaseLayout)/_home/README.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on app/[language]/(BaseLayout)/index.html:
Hi Enn~
Next.js 的 app route 不會放 html 的檔案唷～

enntang replied:
已刪除

JohnsonMao commented on docs/learning-marathons-refactoring-plan.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on docs/home-page-refactoring-plan.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on docs/nextjs15-app-router-guide.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on docs/tailwind-rem-px-reference.md:
Hi Enn~
markdown 檔案可以不要 commit 上來嗎～

enntang replied:
已標記刪除，且設定GIT忽略*.md

JohnsonMao commented on components/ui/button.tsx:
```diff
@@ -35,11 +35,18 @@ const buttonVariants = cva(
+        ctaOrangeSmall:
+          'border-2 border-tips bg-tips text-white hover:!bg-white hover:!text-tips transition-colors duration-200 px-6 py-2 text-sm font-medium rounded-full',
+        ctaPrimary:
+          'border-2 border-primary-base bg-primary-base text-white shadow-[0_8px_10px_0_rgba(22,185,179,0.2)] hover:!bg-white hover:!text-primary-base hover:-translate-y-0.5 hover:shadow-[0_12px_20px_0_rgba(22,185,179,0.3)] active:translate-y-0',
```
在做按鈕 hover 偏移的動態效果時，很容易觸發抖動甚至卡死的 bug
[操作影片](https://jam.dev/c/573f777d-7e08-4711-b277-a4a31f2ee842)

enntang replied:
已移除所有按鈕的hover偏移效果

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/FloatButtons.tsx:
```diff
@@ -0,0 +1,63 @@
+            height={90}
+            className="w-full h-full object-contain animate-spin-slow"
+          />
+        </button>
```
幫我換成 `Button` 的元件，而不是 `button`

enntang replied:
已替換

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/MobileMenu.tsx:
```diff
@@ -0,0 +1,106 @@
+              )}
+            >
+              <span>{item.label}</span>
+            </button>
```
幫我換成 `Button` 的元件，而不是 `button`

enntang replied:
已替換

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/Navbar.tsx:
```diff
@@ -0,0 +1,79 @@
+          aria-label="回到首頁"
+        >
+          <Image src="/assets/landing-page/logo-simple.svg" alt="回到首頁" width={142} height={24} />
+        </button>
```
幫我換成 `Button` 的元件，而不是 `button`

enntang replied:
已替換

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/Navbar.tsx:
```diff
@@ -0,0 +1,79 @@
+            className="text-primary-darker bg-none border-none p-0 font-medium text-base cursor-pointer transition-all duration-300 ease-in-out hover:text-primary-base hover:-translate-y-0.5 relative after:content-[''] after:absolute after:-bottom-1 after:left-0 after:w-0 after:h-0.5 after:bg-primary-base after:transition-all after:duration-300 after:ease-in-out hover:after:w-full"
+          >
+            解決困境
+          </button>
```
幫我換成 `Button` 的元件，而不是 `button`

enntang replied:
已替換

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/Navbar.tsx:
```diff
@@ -0,0 +1,79 @@
+            className="text-primary-darker bg-none border-none p-0 font-medium text-base cursor-pointer transition-all duration-300 ease-in-out hover:text-primary-base hover:-translate-y-0.5 relative after:content-[''] after:absolute after:-bottom-1 after:left-0 after:w-0 after:h-0.5 after:bg-primary-base after:transition-all after:duration-300 after:ease-in-out hover:after:w-full"
+          >
+            功能生態
+          </button>
```
幫我換成 `Button` 的元件，而不是 `button`

enntang replied:
已替換

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/Navbar.tsx:
```diff
@@ -0,0 +1,79 @@
+            className="text-primary-darker bg-none border-none p-0 font-medium text-base cursor-pointer transition-all duration-300 ease-in-out hover:text-primary-base hover:-translate-y-0.5 relative after:content-[''] after:absolute after:-bottom-1 after:left-0 after:w-0 after:h-0.5 after:bg-primary-base after:transition-all after:duration-300 after:ease-in-out hover:after:w-full"
+          >
+            方案
+          </button>
```
幫我換成 `Button` 的元件，而不是 `button`

enntang replied:
已替換

JohnsonMao commented on app/[language]/(BaseLayout)/_home/KeyVision/LottieHero.tsx:
```diff
@@ -0,0 +1,86 @@
+}: Props) {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const animRef = useRef<AnimationItem | null>(null);
+  const [isMobile, setIsMobile] = useState<boolean>(() => window.innerWidth < breakpoint);
```
可以使用 `useMediaQuery` 來判斷裝置的大小，就可以不需要在下方的 useEffect 手動 `setState` 了

enntang replied:
已使用 useMediaQuery 判斷裝置大小

JohnsonMao commented on app/[language]/(BaseLayout)/_home/KeyVision/LottieHero.tsx:
```diff
@@ -0,0 +1,86 @@
+      // 你可以用 CSS 控寬高，這裡放個最小高度避免版面塌陷
+      style={{ minHeight: 240 }}
+    />
+  );
```
可以使用 `min-h-[240px]` 來取代 style 的部分

enntang replied:
已使用Tailwind CSS  `min-h-[240px]`

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Loader/Loader.tsx:
這個寫法其實挺好效能的，而且應該不需要 lottie 就能簡單做到這個效果，另外建議可以透過監聽真實圖片載入來跑進度條，有的設備網路可能比較弱，3.5 秒時間有些圖片還是沒顯示
[操作影片](https://jam.dev/c/91ee8b5e-45f0-47fb-ba96-3615a08b7441)

可以參考我 3 年前做的網站XD，你會發現第一次進網站 loading 比較久，之後就變很快
[https://johnsonmao.github.io/scrum_f2e/](https://johnsonmao.github.io/scrum_f2e/)

enntang replied:
好厲害的作品QQ
我有先靠AI幫我寫一個hook追蹤有標pre-load的圖片，替換掉原本固定3.5s的設定，但是Lottie動畫我還是保留（因為那張Logo圖片是Lottie😅)
現在的問題：我自己 run dev 的時候每次下面的％數都怪怪的（跑不到100%就消失）不知道能不能請大大再幫我看一下orz

JohnsonMao replied:
看起來是判斷載入圖片有寫出 bug，但要花一點時間排查，而且應該只要判斷第一眼會看到的圖片就好了～
之後會優化圖片的格式，改成 webp 的方式，
再麻煩幫我把所有 img, Image 都使用 `components/ui/image` 的元件，這樣後續優化會比較方便

enntang replied:
現已檢查整個專案，`<img>` 標籤都替換為 `<Image/>` 元件

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Footer/Footer.tsx:
```diff
@@ -0,0 +1,88 @@
+        <div className="mb-8">
+          <Image 
+            src="/assets/landing-page/logo-simple-white.svg" 
+            alt="" 
```
需要添加 alt 描述圖片，來確保 SEO 可以知道這是什麼圖片

enntang replied:
已添加描述 `alt="島島阿學 Logo" `

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Footer/Footer.tsx:
```diff
@@ -0,0 +1,88 @@
+                訂閱
+                <ArrowRightIcon />
+              </Button>
+            </form>
```
這個後續看怎麼處理訂閱電子報

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Functions/FunctionCarousel.tsx:
可以使用共用元件 `components/ui/carousel.tsx` 來避免重複造輪子，也不好後續維護

enntang replied:
已改為使用共用元件carousel.tsx

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/MobileMenu.tsx:
```diff
@@ -0,0 +1,106 @@
+
+  return isMobile;
+}
+
```
useIsMobile 改成使用 useMediaQuery 來判斷裝置大小

enntang replied:
已改成使用 useMediaQuery

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/MobileMenu.tsx:
```diff
@@ -0,0 +1,106 @@
+    { label: '解決困境', targetId: 'feature' },
+    { label: '功能生態', targetId: 'functions' },
+    { label: '方案', targetId: 'plans' },
+  ];
```
Enn 幫我把 menuItems 移動到元件外面，避免 useEffect 無謂的觸發

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Navigation/Navbar.css:
這個 css 檔案可以移除囉～

enntang replied:
已移除

JohnsonMao commented on components/ui/arrow-right-icon.tsx:
幫我把 icon 類的 svg 放到 `public/assets/icons` 資料夾下，並且改成 svg 並用 svg 的語法，未來會把圖片部署到 R2 上面降低 Server 的資源負擔

enntang replied:
已移到 /assets/icons/ 並且在components中直接使用 SVG 檔案

JohnsonMao commented on components/ui/arrow-up-icon.tsx:
幫我把 icon 類的 svg 放到 `public/assets/icons` 資料夾下，並且改成 svg 並用 svg 的語法，未來會把圖片部署到 R2 上面降低 Server 的資源負擔

enntang replied:
已移到 /assets/icons/ 並且在components中直接使用 SVG 檔案

JohnsonMao reviewed (changes requested):
很讚噎～超漂亮又完整的，在幫我把留言的部分修正，應該就差不多了～
之後我應該會再調整一些架構和 Testimonials.css 的部分，
然後看小許那邊有 API 可以串接的話，可能會再麻煩你幫忙一起串接 API～

enntang committed:
refactor(BaseLayout): remove unused components and update .gitignore for markdown files

enntang committed:
refactor(BaseLayout): replace button elements with Button component

enntang committed:
fix: resolve ESLint errors in LottieHero and usePreloadImages

- Remove unused className prop from LottieHero component
- Fix consistent-return issues in both files
- Replace ++ operators with += 1 in usePreloadImages
- Replace any types with void in event handlers
- Replace Math.pow with exponentiation operator
- Remove className prop usage from Keyvision.tsx

enntang committed:
refactor(Navbar): add data-preload attribute to logo image for improved performance

enntang committed:
refactor(BaseLayout): replace Arrow icons with Icon component and streamline FunctionCarousel implementation

enntang committed:
refactor(BaseLayout): update FunctionCarousel imports and replace custom mobile detection with media query hook in MobileMenu

JohnsonMao commented on app/[language]/(BaseLayout)/SectionHeader.tsx:
```diff
@@ -0,0 +1,109 @@
+      subtitle: 'text-sm',
+      spacing: 'mb-20',
+    },
+  };
```
目前看有使用 SectionHeader 這個元件，
只有傳遞 light, dark, primary，default 這個類別沒用到，而且 variantStyles 內的 title 與 subtitle 都是用同一個顏色，dark 與 primary 也都是同一個顏色，因此可以簡化，減少冗余的程式碼，例如：
```ts
  const variantStyles = {
    light: 'text-white',
    dark: 'text-primary-darker',
  };
  
  // 然後原本使用傳遞 primary 的改成傳遞 dark 就可以了
```

此外 sizeStyles 也定義了很多冗余的程式碼
目前只有傳遞 md 與 lg，所以可以移除沒用到的 sm 與 xl，
然後又有重複定義同樣的大小，所以 title 與 subtitle 也不需要，例如：
```ts
  const sizeStyles = {
    md: {
      spacing: 'mb-12',
    },
    lg: {
      spacing: 'mb-16',
    },
  };
  
  // 然後在 h2 的 className 多加 text-[1.75rem]
  // h3 的 className 多加 text-sm
  // 字型大小建議用 rem 為單位，對設備響應式會比較好
```

enntang replied:
已整併 SectionHeader 重複的樣式設定

JohnsonMao commented on app/[language]/(BaseLayout)/_home/BubbleDialog/BubbleDialog.tsx:
```diff
@@ -0,0 +1,204 @@
+  const [currentText, setCurrentText] = useState('');
+  const [bubbleWidth, setBubbleWidth] = useState<number | null>(null);
+  const textRef = useRef<HTMLDivElement>(null);
+  const animationRef = useRef<NodeJS.Timeout | null>(null);
```
看起來沒有使用 animationRef ，建議刪除

enntang replied:
已刪除 animationRef

JohnsonMao commented on app/[language]/(BaseLayout)/_home/BubbleDialog/BubbleDialog.tsx:
```diff
@@ -0,0 +1,204 @@
+      // 停止動畫
+      stopAnimation();
+      
+      if (animationRef.current) {
+        clearTimeout(animationRef.current);
+      }
```
animationRef 看起來只有用在 `clearTimeout(animationRef.current);`，
但沒有 setTimeout，所以這段程式碼可以移除。

enntang replied:
已刪除

JohnsonMao commented on app/[language]/(BaseLayout)/_home/BubbleDialog/BubbleDialog.tsx:
```diff
@@ -0,0 +1,204 @@
+      
+      // 使用 requestAnimationFrame 來確保動畫流暢
+      if (isAnimating) {
+        requestAnimationFrame(() => {
+          setTimeout(animateNextLine, 0);
+        });
```
應該可以改成使用 `requestAnimationFrame(animateNextLine)`，不需要特別 `setTimeout`

enntang replied:
已修改為 `requestAnimationFrame(animateNextLine);`

JohnsonMao commented on app/[language]/(BaseLayout)/_home/BubbleDialog/BubbleDialog.tsx:
這個元件裡面的 function 應該要遵守 React 的最佳寫法會比較好，可以參考 (https://ithelp.ithome.com.tw/articles/10306185)[https://ithelp.ithome.com.tw/articles/10306185]，
也可以之後再重構優化

enntang replied:
我有先把 function 搬到元件外面，其他部分因為已經超出理解範圍所以就先沒有動了orz

JohnsonMao commented on app/[language]/(BaseLayout)/_home/CallToAction/CTASection.tsx:
```diff
@@ -0,0 +1,38 @@
+        data-preload
+        className="absolute inset-0 w-full h-full object-cover md:object-contain object-center z-0"
+        aria-hidden="true"
+      />
```
幫我換成 `components/ui/images` 的 `<Image />`，這樣後續再做圖片優化時比較好管理。

enntang replied:
已換成 <Image />

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Presentation/PresentationSection.tsx:
```diff
@@ -0,0 +1,51 @@
+  return (
+    <section className={cn('relative bg-basic-600 pt-16 md:py-24', className)}>
+      {/* 裝飾元素 - 黃色花朵 */}
+      <img 
+        className="absolute top-0 left-[12%] animate-spin-slow" 
+        src="/assets/landing-page/deco-flower-yellow.svg" 
+        alt="裝飾花朵" 
+      />
```
幫我換成 `components/ui/images` 的 `<Image />`，這樣後續再做圖片優化時比較好管理。

enntang replied:
已換成` <Image />`

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Presentation/PresentationSection.tsx:
```diff
@@ -0,0 +1,52 @@
+            alt="學習進度展示" 
+            loading="lazy" 
+            data-preload
+          />
```
幫我換成 `components/ui/images` 的 `<Image />`，這樣後續再做圖片優化時比較好管理。

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Presentation/PresentationSection.tsx:
```diff
@@ -0,0 +1,52 @@
+        className="absolute bottom-0 right-[10%]" 
+        src="/assets/landing-page/deco-mascot-2.svg" 
+        alt="吉祥物裝飾" 
+      />
```
幫我換成 `components/ui/images` 的 `<Image />`，這樣後續再做圖片優化時比較好管理。

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Functions/FunctionCarousel.tsx:
```diff
@@ -0,0 +1,310 @@
+    };
+  }, [getStep, maxIndex, enableNoSnap, disableNoSnap, scrollToIndex]);
+
+  const functions = [
+    {
+      tag: '想法',
+      title: '分享學習洞察',
+      description: '捕捉並分享受到啟發的時刻。在這裡，每個想法都可能點亮別人的學習之路。',
+      imageUrl: 'https://picsum.photos/200/300?grayscale',
+      action: '馬上開始',
+    },
+    {
+      tag: '主題實踐',
+      title: '輕鬆開始學習探索',
+      description: '用 7-30 天的時間嘗試新主題，發現你的興趣。定時打卡，紀錄軌跡和心得！',
+      imageUrl: 'https://picsum.photos/200/300?grayscale',
+      action: '馬上開始',
+    },
+    {
+      tag: '學習計劃',
+      title: '打造你的學習基地',
+      description: '為重要目標建立完整的學習計劃。設定目標、追蹤進度、累積成長！',
+      imageUrl: 'https://picsum.photos/200/300?grayscale',
+      action: '馬上開始',
+    },
+    {
+      tag: '資源',
+      title: '發現與分享學習資源',
+      description: '探索社群推薦的優質學習資源，分享你用過的好內容，並留下真實使用心得。',
+      imageUrl: 'https://picsum.photos/200/300?grayscale',
+      action: '馬上開始',
+    },
+  ];
```
幫忙把 functions 移動到最上層，之後有可能會拉出去用打 API 或靜態資源的方式取得

enntang replied:
已將 functions 移出

JohnsonMao commented on app/[language]/(BaseLayout)/_home/PersonalityTest/PersonalitySection.tsx:
```diff
@@ -0,0 +1,75 @@
+        </Link>
+        </div>
+
+        <div className="absolute bottom-20 left-1/2 transform -translate-x-1/2" />
```
這個 div 應該可以刪除～沒用到

enntang replied:
已刪除

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Testimonials/TestimonialMarquee.tsx:
```diff
@@ -0,0 +1,106 @@
+      name: 'Will',
+      role: '日語學習者 @ 島島阿學',
+      content: '原本總是半途而廢的我，透過主題實踐功能，成功完成了 30 天的日語學習挑戰。',
+      avatar: '/assets/landing-page/avatar-boy.svg',
```
testimonials 可以幫我移動到最頂層，這樣之後這邊的資料可以換成打 API 的方式或抽成靜態資源引入

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Testimonials/TestimonialMarquee.tsx:
```diff
@@ -0,0 +1,106 @@
+    },
+    {
+      name: 'Leo',
+      role: '日語學習者 @ 島島阿學',
+      content: '原本總是半途而廢的我，透過主題實踐功能，成功完成了 30 天的日語學習挑戰。',
+      avatar: '/assets/landing-page/avatar-boy.svg',
+    },
+    {
+      name: 'Enn',
+      role: '設計師 @ 島島阿學',
+      content: '把學習拆進日常情境的做法，對我超實用。',
+      avatar: '/assets/landing-page/avatar-girl.svg',
+    },
+  ];
+
+  return (
+    <section className={cn('section-block marquee-wrap', className)}>
+      {/* 第一排（往右） */}
+      <div className="marquee-row top" aria-hidden="true">
+        <div className="track">
+          {/* 一份內容 */}
+          {testimonials.map((testimonial) => (
+            <TestimonialCard
+              key={testimonial.name}
+              name={testimonial.name}
+              role={testimonial.role}
+              content={testimonial.content}
+              avatar={testimonial.avatar}
+            />
+          ))}
+          {/* 再複製一次（無縫循環關鍵） */}
+          {testimonials.map((testimonial) => (
+            <TestimonialCard
+              key={`duplicate-${testimonial.name}`}
+              name={testimonial.name}
+              role={testimonial.role}
+              content={testimonial.content}
+              avatar={testimonial.avatar}
+            />
+          ))}
```
另外可以多增加 id，並使用 uuid 來生成唯一 id

enntang replied:
已修改key寫法並使用uuid

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Testimonials/Testimonials.css:
這個有機會可以整合到 tailwind config 裡面，未來比較好複用樣式

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Videos/VideoItem.tsx:
```diff
@@ -0,0 +1,53 @@
+        controls 
+        className="aspect-video w-full rounded-[20px]"
+      >
+        <source />
```
這個之後有影片在處理

JohnsonMao commented on components/ui/icon.tsx:
這與我想像中的不太一致XD
不過先這樣吧～

JohnsonMao commented on app/[language]/(BaseLayout)/_home/Loader/Loader.tsx:
```diff
@@ -0,0 +1,77 @@
+            className="h-full bg-primary-base transition-[width] duration-200 ease-in-out"
+            style={{ 
+              width: `${Math.min(progress * 100, 100)}%`,
+            }}
```
使用 transform 的 scale 對畫面渲染的效能更友善，另外可以用 `components/ui/progress.tsx` 就可以有進度條的效果了

enntang replied:
已換成 progress 元件

JohnsonMao commented on hooks/usePreloadImages.ts:
loading 的動畫邏輯還是太複雜了XD
後續再重構吧～
另外 `components/ui/image` 的 `<Image />` 有個參數可以讓圖片提早載入（但不可以濫用），僅用在最上面一開始會看到的圖片就好了

JohnsonMao replied:
因為圖片在滾動到畫面之前，不會載入圖片，所以就不會出現，也就不會載入，就會卡在那邊，只需把第一個畫面需要的圖片加上 preload 就好，不用全部圖片，這樣也比較快進入首頁，另外如果可以的話，需要在 loading 的時候鎖住畫面，因為目前 loading 時，其實你還是可以滾動畫面，所以你直接一直往下滾，就會到 100%

enntang replied:
我有在最上面畫面的 Logo 加上 `data-preload` `priority` 這兩個屬性了～第一個是給Loader用的自定義屬性，第二個應該就是你說的 `<Image/>` 專用參數><
然後鎖定捲動的功能我有請 AI 做成 hooks!

JohnsonMao commented on hooks/useScrollVisibility.ts:
這個 hook 沒用到，應該可以移除

enntang replied:
這個有被用於判斷 navbar 和側邊按鈕出現時機喔～><

JohnsonMao commented on tsconfig.tsbuildinfo:
這個幫我移除加入 .gitignore 感謝～

enntang replied:
已加入git ignore 並標記刪除

JohnsonMao reviewed (changes requested):
剩下的這些修正應該就可以先合進去了～
後續再慢慢重構

enntang committed:
refactor(BaseLayout): simplify SectionHeader props and integrate into FeatureGrid

enntang commented on app/[language]/(BaseLayout)/_home/KeyVision/LottieHero.tsx:
```diff
@@ -0,0 +1,86 @@
+      }
+    };
+    // 依賴 isMobile / 路徑變動時重載
+  }, [isMobile, desktopSrc, mobileSrc, breakpoint, preserveAspectRatio]);
```
已在先前的 commit push 修正（使用 useMediaQuery 代替 isMobile state）

enntang commented on app/[language]/(BaseLayout)/_home/Footer/Footer.tsx:
```diff
@@ -0,0 +1,88 @@
+                訂閱
+                <ArrowRightIcon />
+              </Button>
+            </form>
```
已在先前的 commit push 修正

enntang commented on app/[language]/(BaseLayout)/_home/Loader/Loader.tsx:
```diff
@@ -0,0 +1,95 @@
+      // easeOutCubic
+      const eased = 1 - (1 - currentProgress) ** 3;
+      setProgress(eased);
+      console.log('Progress updated:', eased, 'Percentage:', Math.round(eased * 100));
```
已移除

enntang commented on app/[language]/(BaseLayout)/SectionHeader/README.md:
```diff
@@ -0,0 +1,167 @@
+
+| Prop | 類型 | 預設值 | 說明 |
+|------|------|--------|------|
+| `title` | `string` | - | 主要標題文字（必填） |
```
*.md 已列入 git ignore

enntang committed:
refactor(BaseLayout): enhance testimonials with Tailwind CSS and improve marquee functionality

enntang committed:
refactor(BaseLayout): update image elements to use Next.js Image component and adjust styles for better responsiveness

enntang committed:
refactor(BaseLayout): remove data-preload attributes from image elements and enhance Loader component with scroll lock functionality

enntang committed:
refactor(BaseLayout): enhance FunctionCarousel and KeyVision components with improved styling and layout adjustments

enntang commented on app/[language]/(BaseLayout)/_home/Navigation/Navbar.tsx:
```diff
@@ -0,0 +1,79 @@
+        : isVisible 
+        ? 'opacity-100 translate-y-0 pointer-events-auto' 
+        : 'opacity-0 -translate-y-full pointer-events-none'
+    }`}>
```
已簡化邏輯

enntang committed:
refactor(BaseLayout): optimize BubbleDialog animation handling and simplify Navbar visibility logic

enntang committed:
refactor(BaseLayout): implement parallax effects in CTASection, FunctionCarousel, PersonalitySection, and SloganSection for enhanced visual dynamics

enntang committed:
refactor(BaseLayout): update FunctionCarousel and SloganSection for improved parallax handling and code consistency

enntang committed:
refactor(BaseLayout): enhance BubbleDialog  logic

enntang committed:
Remove tsconfig.tsbuildinfo from tracking and add to .gitignore

enntang committed:
Add tsconfig.tsbuildinfo to .gitignore to prevent tracking

JohnsonMao commented on hooks/useParallax.ts:
```diff
@@ -0,0 +1,105 @@
+  configs: Array<UseParallaxOptions & UseParallaxPresetOptions>
+) {
+  return configs.map(config => useParallax<T>(config));
+}
```
這個寫法不安全，如果 configs 有變動的時候會導致 hook 執行順序變動，而產生嚴重的 bug

enntang replied:
想了一下還是先移除這個特效好了XD

上面navbar的狀況
只有捲動到綠色背景+滑鼠hover才會出現，這個還好～

JohnsonMao reviewed (changes requested):
沒想到還加了視差動畫，雖然看起來有點微妙XD

手機版有個段落呈現如下圖，是預期的嗎？
<img width="376" height="132" alt="image" src="https://github.com/user-attachments/assets/3451b5a3-ac7b-4848-85dd-4d0cda66e78e" />

navbar 在滾動到某些區域會不好識別，也是預期的嗎～？
<img width="1067" height="81" alt="image" src="https://github.com/user-attachments/assets/37582d0b-a764-4247-9767-1cde99334fa6" />

確認一下而已～

等 useMultipleParallax 修正後應該就會合進去了，
對了～還有衝突需要解一下

enntang committed:
refactor(BaseLayout): remove parallax effects

JohnsonMao committed:
Merge branch 'dev' into feat/landing-page

JohnsonMao committed:
chore(ci): remove pnpm version constraint in workflows

JohnsonMao committed:
Merge branch 'dev' into feat/landing-page

JohnsonMao reviewed (commented):
LGTM
